pkg/volume/flexvolume/plugin.go

```diff
@@ -177,6 +177,14 @@ func (plugin *flexVolumePlugin) newMounterInternal(spec *volume.Spec, pod *api.P
 		return nil, err
 	}
 
+	var metricsProvider volume.MetricsProvider
+	if plugin.capabilities.SupportsMetrics {
```

Comment:
Is this accurate? Since flex is also used for ephemeral storage, any plugin that says it supports is going to default to statfs based implementation. The problem with that is - for ephemeral storage, the storage space could be shared with the host and hence statsfs likely will not return accurate information. Same as CSI, I am afraid this information has to come from plugin itself.

In fact - this is perhaps more applicable to flex than CSI because there are quite a bit of flex plugins that are used for ephemeral data/tokens etc.

Reply:
cc @chakri-nelluri

Reply:
@gnufied If the plugins like ephemeral storage cannot support, can we assume they won't return true for SupportsMetrics? With that assumption, the change LGTM.

Reply:
We can extend this call to Flex driver, but given we are planning to phase out flex in favor of CSI, I would hold off extending it unless there is a real push asking for it.

Reply:
okay, lets get this shipped. Does not sound like a blocker.